benchmark/src/submit_pattern.rs

```diff
@@ -0,0 +1,87 @@
+use logger::prelude::*;
+
+/// Consume constant amount of items in every time interval.
+/// Flood all available items when constant = usize::MAX.
+#[derive(Debug, Clone)]
+pub struct ConstantRate {
+    /// Constant submission rate per second.
+    constant: usize,
+    /// Submission interval in milliseconds. Default 1000 ms.
+    pub(crate) interval_ms: usize,
+    /// Number of items to drain in a interval of interval_ms milliseconds.
+    amount: usize,
+}
+
+impl ConstantRate {
+    pub fn new(constant: usize) -> Self {
+        ConstantRate {
+            constant,
+            interval_ms: 1000,
+            amount: constant,
+        }
+    }
+
+    /// Scale down the submission interval when the amount of items to consume can divide 10.
+    /// For example, for 100 req / s, return (1 req, 10 ms).
+    fn adjust_submission_interval_ms(&mut self) {
+        while self.amount % 10 == 0 && self.interval_ms > 10 {
```

Comment:
Why do you need to make this calculation here? Can you set a finer granularity from the get go? e.g., how many micros to wait between sending a new transaction?
It might be easier than expecting everything to always divide by 10.